### PR TITLE
Reworks and Reintroduced Micro Reactor & Radiation Scrubber CBM's

### PR DESCRIPTION
**Pull request by MrLostman**
#### Summary
SUMMARY:[Content] "Reworks micro reactor & radiation CBM's and adds the reactors back to the drop table"

#### Purpose of change
To fill a gap in high tech power generation CBM's

#### Describe the solution
Edit : a * represents a revision edit

Micro Reactor
- dropped from husks & shocker brutes. low drop from cyborgs & sci body sites
- 5 chest encumbrance
- 1 plutonium cell gives 2500 charges @ 10kw
- holds 4 cells
- 10 irradiation stacks a day *only while fueled
- *produces a constant ~~500 J~~ 50 J of power while fueled for no cost

Advanced Micro Reactor 
- dropped from bank vaults
- 5 chest encumbrance
- 1 plutonium cell gives 3000 charges @ 12kw
- holds 4 cells
- 5 irradiation stacks a day *only while fueled
- *produces a constant ~~1 kJ~~ 100J of power while fueled for no cost

Radiation Scrubber CBM
~~- lowers radiation produced by the Micro Reactor by half and completely stops radiation of the Advanced Micro Reactor~~
- *reworked to be toggled removing up to 10 rads so one stack every 10 mins
- *10 J to have running and a cost of 10 kJ only when it removes rads

*Reactor Safety Override
- comes with the normal and advanced reactors
- produces an extra 40kJ of power a turn
- 0.5 radiation stacks a turn

#### Testing
1. checked functionality by spawning in the cbm's and testing
2. checked spawns
3. checked natural instillation & upgrade
4. passed time with all combinations to test irradiation

### CODE REVIEW

**Comment by chaosvolt** [comment]
Should the radiation be constant, even when not supplied with fuel or otherwise in use? Seems more fitting for it to apply bursts of radiation when used, to properly make it an "oh shit" powergen option with consequences when the situation is dire enough to rely on.

**Comment by MrLostman** [comment]
Just put this on hold for a little bit, going to rework the rework radiation scrubber too and add another setting to the reactor

**Comment by scarf005** [comment]
would love the reactor to perpetually generate small amount of energy on default like IRL.

File: tests/bionics_test.cpp
Comment (scarf005) [inline tests/bionics_test.cpp:1]: could you please explain the reason for removing the test code? adding more tests would help verify the new feature correctly works.

**Review by scarf005 — COMMENTED** [review]
also, using [descriptive commit names](https://dev.to/samuelfaure/how-atomic-git-commits-dramatically-increased-my-productivity-and-will-increase-yours-too-4a84) makes it clear what the commit does and helps both author and reviewer identify changes quickly.

**Review by chaosvolt — COMMENTED** [review]
1. Compiled and load-tested.
2. Installed a ton of power storage CBMs before adding micro-reactor.
3. Waited a while, observed that no radiation or power generation occured without fuel.
4. Loaded a single plutonium cell in, waited a while to confirm power generation and radiation occured.
5. Turned it on, observed that power generation is 10 kj per turn.
6. Debugged in the upgrade CBM and installed it, confirmed it converts the normal reactor to advanced, and that you still only have one copy of the override CBM on you.

I'm not yet 100% sure about the amount of passive powergen it gives, since you can get the same amount of power you'd get out of burning that cell just by waiting a lil over an hour, and even faster for the advanced version. I'd be tempted to suggest 50 and 100 watts for the passive powergen instead, that leaves it still a constant source of free energy but makes the choice to burn it a much more tantalizing prospect in comparison.

**Review by scarf005 — APPROVED** [review]
looks good. could you try JSONizing persistent powergen for reactors sometimes later? or would it be unneccesary because there's few permagen CBMs?

**Pull request merged**